webui/src/lib/components/controls.jsx

```diff
@@ -274,6 +275,56 @@ export const ClipboardButton = ({ text, variant, onSuccess, icon = <PasteIcon/>,
     );
 };
 
+export const PrefixSearchWidget = ({ onFilter, text = "Search by Prefix", defaultValue = "" }) => {
+
+    const [expanded, setExpanded] = useState(!!defaultValue)
+
+    const toggle = useCallback((e) => {
+        e.preventDefault()
+        setExpanded((prev) => {
+            return !prev
+        })
+    }, [setExpanded])
+
+    const handleSubmit =(e) => {
```

Comment:
Why not add a `useCallback` here? The idea of `useCallback` is to memoize the function instance/reference so that it doesn't change between renders. If you pass the function instance as a prop and the ref changes, the component will re-render. Usually, React does this fast enough, but these things tend to pile on.

Reply:
Right! Added